crawl-ref/source/l-travel.cc

```diff
@@ -112,6 +112,28 @@ LUAFN(l_waypoint_delta)
     return 2;
 }
 
+/*** Set a numbered waypoint.
+ * Uses player-centered coordinates.
+ * @tparam int waynum
+ * @tparam int x
+ * @tparam int y
+ * @function set_waypoint
+ */
+LUAFN(l_set_waypoint)
+{
+    int waynum = luaL_safe_checkint(ls, 1);
+    if (waynum < 0 || waynum > 9)
+        return 0;
```

Comment:
possibly print an error of some kind here with `luaL_error`.

Reply:
done; also added luaL_errors to other bounds checks